lib/account-data.js

```diff
@@ -1,95 +1,159 @@
 const fs = require('fs')
 const path = require('path')
 const rimraf = require('rimraf')
-const os = require("os");
-const {data,appDataPath} = require('./config')
-const userFiles = path.join(os.homedir(),'AppData','Roaming')
-const parsecUser = path.join(userFiles,'Parsec','user.bin')
-const accountData = path.join(appDataPath,'account_data.json')
-const currentAccountFile = path.join(appDataPath,'currentAccount.txt')
-
-const multiUserFolder = path.join(userFiles,'Parsec','UserList')
-
+const {global_state,initialize} = require("./initialize")
 
+const {fileFuncs} = require("./file-operations")
+// initialize()
 //todo need some rewriting.
 
-
 function checkNickname(nickname){
+
     let exist = 0
-    if(data.has(nickname)){
+    if(global_state.data.has(nickname)){
         exist = 1
     }
     return exist
 }
 
-function removeParsecUserFile(){
-    if (fs.existsSync(parsecUser)){
-        fs.unlinkSync(parsecUser)
+async function removeParsecUserFile(){
+    if (fs.existsSync(global_state.parsecUser)){
+        let op = await fileFuncs.deleteFile(global_state.parsecUser)
+        if (!op.success){
+            return 5
+        }
+        else {
+            return 0
+        }
+        //fs.unlinkSync(global_state.parsecUser)
     }
+    return 0
 
 
 }
-function backupParsec(){
-
-}
+// function backupParsec(){
+//
+// }
+//
+// function createAccount(nickname){
+//
+//
+//     let newUserFolder = path.join(global_state.multiUserPath,nickname)
+//     let newUserFile = path.join(newUserFolder,'user.bin')
+//     let fs.mkdirSync(newUserFolder)
+//
+//     fs.copyFileSync(global_state.parsecUser,newUserFile)
+//
+//     global_state.data[nickname] = newUserFile
+// }
+
+async function removeAccount(nickname){
+
+    delete global_state.data[nickname]
+
+
+
+    let userPathWithoutBin = path.join(global_state.multiUserPath,nickname)
+    let userPath = path.join(global_state.multiUserPath,nickname,'user.bin')
+    if(fs.existsSync(userPath)){
```

Comment:
use async/await alternative if that exists, it'll be better for GUI

Reply:
done